### PR TITLE
Use GitHub Actions for Pull Requests

### PR DESCRIPTION
**Pull request by olix0r**
Disables Travis on Pull Requests, replacing it with GitHub Actions.
Travis remains enabled for master builds for the moment.

### CODE REVIEW

File: .github/workflows/pull_request_rust.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+name: Pull Request (Rust)
+
+on:
+  pull_request:
+    paths:
+    - 'Cargo.*'
+    - '*.rs'
```
Comment (hawkw) [inline .github/workflows/pull_request_rust.yml]: This is extremely cool! :D

File: .github/workflows/pull_request_rust.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+name: Pull Request (Rust)
+
+on:
+  pull_request:
```
Comment (hawkw) [inline .github/workflows/pull_request_rust.yml]: Will this be re-run for every push to a PR,  or just run once when the PR is created?

Comment (olix0r) [reply .github/workflows/pull_request_rust.yml]: all pushes to a pr, it seems

Comment (hawkw) [reply .github/workflows/pull_request_rust.yml]: 👍

**Comment by olix0r** [comment]
I'm a bit miffed as to why the rust build is running on this PR and the docker build is not, but we'll fix it in post!

**Pull request merged**